development/extensions/index.md

```diff
@@ -284,9 +355,169 @@ Once installed on the [Onboard Computer](@/hardware/required/onboard-computer/in
+
+TODO: describe how to register an existing image as an Extension, by "installing" a custom extension with a tag that doesn't exist online
+(in which case it should fall back to the existing offline image with that tag).
+
+{% note() %}
+While this can be useful for testing minor changes during development (e.g. fixing a small bug, or adjusting some logging outputs), it is
+not recommended to upload commit-modified images as new Extension versions, because there's then no record in the source code repository of the change(s).
+{% end %}
+
 {% note() %}
-Walkthrough coming soon!
+If you want to make changes to files that are mounted to the computer's filesystem (via `binds` in the `permissions` metadata LABEL) then that can instead
+be done via the [File Browser](../../advanced-usage/#file-browser), restarting the Extension afterwards if necessary. In that case the changes will be
+persistent once the modified file has been saved.
 {% end %}
 
-- [Sneak peek diagram](https://github.com/bluerobotics/ardusub-zola/blob/BlueOS-1.1/development/extensions/development-process.mermaid)
+-->
+
+#### Submission to the Bazaar
+
+1. Check [metadata docs](#metadata-dockerfile), and match the requirements with your Dockerfile
+1. Build, and upload a [SemVer](https://semver.org/)-compliant Docker Image to Docker Hub
+1. Submit a Pull Request (PR) to the [extensions repo](https://github.com/bluerobotics/BlueOS-Extensions-Repository)
+    - Include an icon, and a user-friendly description
+    - A nicer form-based interface is on the way, but not yet available
+
+#### Getting Your Extension Accepted
+
+1. Check for build errors in the Docker Image creation process
+1. Respond to PR feedback (if there is any)
+1. Blue Robotics will merge the PR, after which it becomes public
+    - Available extensions are visible in the BlueOS extensions manager, as well as in the [gallery](https://docs.bluerobotics.com/BlueOS-Extensions-Repository/)
+    - BlueOS Bazaar will provide a filterable and searchable interface for this
+
+#### Installing from the Bazaar
+
+The Bazaar is available through the BlueOS [Extensions Manager](../../advanced-usage/#extensions-manager). Once an Extension has been accepted into the Bazaar it should appear automatically, within a few minutes.
+
+#### Iterating
+
+Most programming projects are never fully "complete", so it is expected that Extensions will be modified and improved over time. Extension approval only needs to occur when the Extension is first added to the store, so updated releases should become available within 24 hours from when the new version is uploaded to Docker Hub.
+
+The expected iteration process is:
+
+1. Make improvements
+1. Build the updated Docker Images and upload them to Docker Hub
+    - Should be tagged with a SemVer compliant tag that’s higher than the previous release
+        - E.g. v1.0.0 → v1.0.1
+1. Confirm the new version is available through the BlueOS Extensions Manager
+1. Install and test it
```

Comment:
That should be a release cycle and not a development one.
At the moment, I believe that we don't have a friend way of doing such thing. The user would need to run their docker manually from the raspberry and upload custom tags (that are not semver) to test it. The reason behind this is that any semver tag will be available on the store currently, but I believe that is not the case once we move to blueos.cloud / bazaar. @voorloopnul  ?

Reply:
Releases are part of the application development, although I do agree that "Release Cycle" is a good name for this part.

On the "friendly development cycle" side of things, I think that will require a combination of:
1. bluerobotics/BlueOS#1614
2. Custom tag upload and installation through the extensions manager
3. Locally modifying existing containers, giving them a custom tag, and doing offline registration of the generated image, as Willian and I are working on documenting in aedf6182d88aa10a4c4426ac8266c41f15e3465e (which is in this PR, but currently not shown in the built docs because the instructions are incomplete)